src/components/StarRating/index.tsx

```diff
@@ -0,0 +1,100 @@
+import { ReactComponent as Star } from 'assets/svg/star.svg';
+import {
+  createContext, useMemo, useState,
+} from 'react';
+import useContext from '../../utils/hooks/useContext';
+import styles from './StarRating.module.scss';
+
+interface StarRateContextType {
+  enter: () => void;
+  leave: () => void;
+  click: (num: number) => void;
```

Comment:
아래 이름이 더 좋지 않을까 싶어요 핸들러를 내려주는 거니깐요

```suggestion
  handleMouseEnter: () => void;
  handleMouseLeave: () => void;
  handleClick: (num: number) => void;
```